compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -149,7 +139,21 @@ class ExtractDependencies extends Phase {
       def allowLocal = dep.context == DependencyByInheritance || dep.context == LocalDependencyByInheritance
       if (depFile.extension == "class") {
         // Dependency is external -- source is undefined
-        processExternalDependency(depFile)
+
+        // For non-local classes this should be equivalent to `atPhase(flattenPhase) { dep.to.fullName }` but
+        // faster as it doesn't require running any denotation transformer.
+        val binaryClassName = {
+          val builder = new StringBuilder
+          val pkg = dep.to.enclosingPackageClass
+          if (!pkg.isEffectiveRoot) {
+            builder.append(pkg.fullName.mangledString)
+            builder.append(".")
+          }
+          builder.append(dep.to.flatName.mangledString)
```

Comment:
this doesnt work for java module class

Reply:
Yes, about to push a fix, thanks for noticing :).